engine/withdraw.go

```diff
@@ -0,0 +1,102 @@
+package engine
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	withdrawal "github.com/thrasher-corp/gocryptotrader/database/repository/withdraw"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+const (
+	// ErrWithdrawRequestNotFound message to display when no record is found
+	ErrWithdrawRequestNotFound = "%v not found"
+	// ErrRequestCannotbeNill message to display when request is nil
+	ErrRequestCannotbeNill = "request cannot be nil"
+)
+
+// SubmitWithdrawal preforms validation and submits a new withdraw request to exchange
+func SubmitWithdrawal(exchName string, req *withdraw.Request) (*withdraw.Response, error) {
+	if req == nil {
+		return nil, errors.New(ErrRequestCannotbeNill)
+	}
+	if req.Exchange == "" {
+		req.Exchange = exchName
+	}
+	err := withdraw.Valid(req)
+	if err != nil {
+		return nil, err
+	}
+	exch := GetExchangeByName(exchName)
+	if exch == nil {
+		return nil, ErrExchangeNotFound
+	}
+
+	resp := &withdraw.Response{
+		Exchange: &withdraw.ExchangeResponse{
+			Name: exchName,
+		},
+		RequestDetails: req,
+	}
+
+	if Bot.Settings.EnableDryRun {
+		log.Warnln(log.Global, "Dry run enabled no withdrawal request will be submitted or event created")
```

Comment:
Grammar change: "Dry run enabled, no withdrawal request will be submitted or have an event created"